### PR TITLE
Added test id for menu

### LINKED ISSUE #1015: Add data-cy label for Menu component
- https://github.com/bigbinary/neeto-chat-web/commit/9923465312546a4a2a952aa1d1c50182369eeb8d

### PR DESCRIPTION
**Pull request by SreehariJayaraj**
- Fixes #1015 

**Description**

**Checklist**

- [ ] I have made corresponding changes to the documentation.
- [ ] I have updated the types definition of modified exports.
- [ ] I have verified the functionality in some of the neeto web-apps.
- [ ] I have added the necessary label (patch/minor/major - If package publish
      is required).
      
      
      
Screenshot 2024-03-07 at 2 09 47 PM

I have added data cy labels for all the menu buttons

**Reviewers**

### CODE REVIEW

**Review by VarunSriram99 — CHANGES_REQUESTED** [review]
@sreehari2003 A few questions,

**Questions**
- Why are we going through all these steps for adding a data-cy label? Button has option for other props. So just add the "data-cy" label in the buttonProps?
- Why are we adding a data-test-id label when we use data-cy labels in our tests?
- Even if you're adding a test-id, it should `data-testid` and not `data-test-id`
- Why are we not using the hyphenize utlity from neetoCist?

**Comment by SreehariJayaraj** [comment]
> @sreehari2003 A few questions,
> 
> **Questions**
> 
> * Why are we going through all these steps for adding a data-cy label? Button has option for other props. So just add the "data-cy" label in the buttonProps?
> * Why are we adding a data-test-id label when we use data-cy labels in our tests?
> * Even if you're adding a test-id, it should `data-testid` and not `data-test-id`
> * Why are we not using the hyphenize utlity from neetoCist?

Screenshot 2024-03-09 at 8 41 47 PM

hyphenate is not available to import

File: src/components/Editor/Menu/Headless/Option/index.jsx [hunk 1]
```diff
@@ -27,7 +27,8 @@ const Option = ({
   return (
     <Button
       className={isActive ? "ne-headless-btn--active" : ""}
-      {...{ disabled }}
+      {...{ disabled, optionName }}
+      data-cy={optionName.replace(/\s+/g, "-")}
```
Comment (VarunSriram99) [inline src/components/Editor/Menu/Headless/Option/index.jsx:31]: Use [hyphenate from neeto-cist ](https://github.com/bigbinary/neeto-cist/blob/main/docs/pure/strings.md#hyphenate).

**Review by VarunSriram99 — CHANGES_REQUESTED** [review]
@sreehari2003 _a Please make sure you're taking care of all the review suggestions before re-assigning.

**Comment by VarunSriram99** [comment]
> > @sreehari2003 A few questions,
> > **Questions**
> > 
> > * Why are we going through all these steps for adding a data-cy label? Button has option for other props. So just add the "data-cy" label in the buttonProps?
> > * Why are we adding a data-test-id label when we use data-cy labels in our tests?
> > * Even if you're adding a test-id, it should `data-testid` and not `data-test-id`
> > * Why are we not using the hyphenize utlity from neetoCist?
> 
> Screenshot 2024-03-09 at 8 41 47 PM
> hyphenate is not available to import

@sreehari2003 its hyphenate. Please go through all the neeto-cist and Playwright commons utilities carefully to make sure we're not repeating the logic.

**Comment by SreehariJayaraj** [comment]
- This Issue needs to be fixed first  #1027

**Review by VarunSriram99 — APPROVED** [review]
@sreehari2003 _a LGTM

**Pull request merged**